src/sagas/wallet.js

```diff
@@ -637,7 +650,7 @@ export function* walletReloading() {
 export function* refreshSharedAddress() {
   const wallet = yield select((state) => state.wallet);
 
-  const { address, index } = wallet.getCurrentAddress();
+  const { address, index } = yield wallet.getCurrentAddress();
```

Comment:
Any `yield promise` works inside a saga like an await but the [docs](https://redux-saga.js.org/docs/basics/DeclarativeEffects/) prefer using `call` since it simplifies testing.

The same docs page suggests using an alternative way to use call which passes a `this` context (much like bind)
`yield call([wallet, wallet.getCurrentAddress])`
`yield call([wallet, 'getCurrentAddress'])`
`yield apply(wallet, 'getCurrentAddress')` ([apply](https://redux-saga.js.org/docs/api/#applycontext-fn-args) is an alias for the same syntax as above)

All of these work similarly to `yield call(wallet.getCurrentAddress.bind(wallet))` but with a better syntax

Reply:
That's some exotic syntax but it makes perfect sense!
I believe this will be an improvement to be replicated on the mobile wallet as well.